zokrates_github_resolver/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+        let file = File::open(&pb)?;
+        let br = BufReader::new(file);
+
+        let alias = PathBuf::from(path.clone())
+            .as_path()
+            .file_stem()
+            .unwrap()
+            .to_string_lossy()
+            .to_string();
+
+        Ok((br, location.to_owned(), alias))
+    } else {
+        Err(io::Error::new(io::ErrorKind::Other, "No location provided"))
+    }
+}
+
+/// Checks that import source is using github import location.
+pub fn is_github_import(source: &str) -> bool {
+    source.starts_with(GITHUB_IMPORT_PREFIX)
+}
+
+/// Parses github import syntax: "github:<root>/<repo>/<branch>/<path/to/file>"
+/// Where:
+/// - <root> is the user or organization name
+/// - <repo> is the repository name
+/// - <branch> is the branch/snapshot name, e.g. `master`
+/// - <path/to/file> is the absolute path to file in the specified branch of the repository
+fn parse_input_path(path: &str) -> Result<(String, String, String, String), io::Error> {
+    let path = path.replacen("github:", "", 1);
+    if path.contains("..") {
+        return Err(io::Error::new(
+            io::ErrorKind::Other,
+            "Invalid github import syntax. It must not contain '..'",
+        ));
+    }
+
+    let components = path.split("/").collect::<Vec<_>>();
+
+    // Check that root, repo, branch & path are specified
+    if components.len() < 4 {
+        return Err(io::Error::new(
+            io::ErrorKind::Other,
+            "Invalid github import syntax. Should be: github:<root>/<repo>/<branch>/<path>",
+        ));
+    }
+
+    let root = components[0];
+    let repo = components[1];
+    let branch = components[2];
+    let path = components[3..].join("/"); // Collect the rest of the import path into single string
+
+    Ok((
+        root.to_owned(),
+        repo.to_owned(),
+        branch.to_owned(),
+        path.to_owned(),
+    ))
+}
+
+/// Downloads the file from github by specific root (user/org), repository, branch and path.
+fn download_from_github(
+    root: &str,
+    repo: &str,
+    branch: &str,
+    path: &str,
+) -> Result<PathBuf, io::Error> {
+    let url = format!(
+        "https://raw.githubusercontent.com/{root}/{repo}/{branch}/{path}",
+        root = root,
+        repo = repo,
+        branch = branch,
+        path = path
+    );
+
+    Ok(download_url(&url)?)
+}
+
+fn download_url(url: &str) -> Result<PathBuf, io::Error> {
+    let mut response = reqwest::get(url).map_err(|e| {
+        io::Error::new(
+            io::ErrorKind::Other,
+            format!("Unable to access github: {}", e.to_string()),
+        )
+    })?;
+
+    let (mut dest, pb) = NamedTempFile::new()?.keep()?;
+    copy(&mut response, &mut dest)?;
+
+    Ok(pb)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    pub fn import_simple() {
+        let res = parse_input_path("github:eupn/test/master/examples/imports.code").unwrap();
```

Comment:
I think we should have the tests point to the ZoKrates repo here.

We also need to test `resolve` which requires mocking github for determinism and to avoid requiring an internet connexion to run the tests. Maybe we could use [mockito](https://docs.rs/mockito/0.17.1/mockito/) 

Any thoughts?

Reply:
Redirected everything to ZoKrates repo. I can try to gate some kind of `reqwest` mock via `#[cfg(test)]` though.

Reply:
That would be very nice!